app/src/main/java/com/aileyzhang/musicapp/SongData.java

```diff
@@ -54,15 +48,52 @@ private void loadAllSongs(Context context) {
                     Bitmap art = BitmapFactory.decodeStream(inputStream);
 
                     Song song = new Song(path, title, artist, album, art);
-                    mSongs.add(song);
+                    songs.add(song);
                 }
             }
+            return songs;
         } finally {
             if (songCursor != null) songCursor.close();
         }
     }
 
-    private void loadSongsFromAlbum(String albumID) {
+    public static ArrayList<Song> getSongsInAlbum (Context context, String albumName) {
+        ArrayList<Song> songs = new ArrayList<>();
+        String[] projection = new String[]{
+                MediaStore.Audio.AudioColumns.DATA,
+                MediaStore.Audio.AudioColumns.TITLE,
+                MediaStore.Audio.ArtistColumns.ARTIST,
+                MediaStore.Audio.AudioColumns.ALBUM,};
+        String selection = MediaStore.Audio.AudioColumns.ALBUM + "=?";
+        String[] selectionArgs = new String[] {"" + albumName};
```

Comment:
Why is there a `"" + <somethingThatsAlreadyAString>`, and also do we need to query by albumName? Does album ID not work or something? =w=

Reply:
I'll change that. albumID doesn't work. Think it's unique for the same albums as well. The cursor returns null for me when I use albumID